### PR TITLE
chore(deps): stylable@4, webpack@5, latest others

### PR DESCRIPTION
**Pull request by AviVahl**

### CODE REVIEW

**Comment by AviVahl** [comment]
passes typechecking and lint.
currently fails executing features built as external features. need to figure out why.

@RomanYarik

File: packages/scripts/src/create-webpack-configs.ts [hunk 1]
```diff
@@ -276,8 +276,10 @@ export function createWebpackConfigForExteranlFeature({
             path: outputPath,
             filename: `[name].${target}.js`,
             chunkFilename: `[name].${target}.js`,
-            libraryTarget: 'var',
-            jsonpFunction: packageName + name,
+            library: {
```
Comment (AviVahl) [inline packages/scripts/src/create-webpack-configs.ts:279]: this change is probably why the tests are failing. need to figure out how this has to be configured on webpack@5.

File: packages/engineer/src/feature/dev-server.dev-server.env.ts [hunk 2]
```diff
@@ -15,27 +15,17 @@ import {
...
-        return {
-            close(cb?: () => void) {
-                if (cb) {
-                    cb();
-                }
-            },
-            invalidate: () => undefined,
-        };
-    };
+// eslint-disable-next-line @typescript-eslint/no-var-requires
```
Comment (RomanYarik) [inline packages/engineer/src/feature/dev-server.dev-server.env.ts:24]: why? are the types wrong when importing?

File: packages/scripts/src/create-webpack-configs.ts [hunk 3]
```diff
@@ -276,8 +276,10 @@ export function createWebpackConfigForExteranlFeature({
...
+                type: 'global',
```
Comment (RomanYarik) [inline packages/scripts/src/create-webpack-configs.ts:280]: why did you change to 'global'?

**Review by RomanYarik — COMMENTED** [review]
no need for single run anymore? how do you make the webpackDevMiddeware to "not watch" for current run?

File: packages/scripts/src/create-webpack-configs.ts (hunk 3 above)
Comment (AviVahl) [reply packages/scripts/src/create-webpack-configs.ts:280]: got a runtime error when using type: `jsonp` with that name. maybe I'm missing something obvious.

File: packages/engineer/src/feature/dev-server.dev-server.env.ts (hunk 2 above)
Comment (AviVahl) [reply packages/engineer/src/feature/dev-server.dev-server.env.ts:24]: known issue. anything that imports `@types/webpack` is currently broken. there are discussions about this in DefinitelyTyped.

File: packages/scripts/src/create-webpack-configs.ts (hunk 3 above)
Comment (RomanYarik) [reply packages/scripts/src/create-webpack-configs.ts:280]: from what I read at https://webpack.js.org/blog/2020-10-10-webpack-5-release/#automatic-unique-naming I see that the `jsonpFunction` is no longer needed.
maybe just restore the `libraryTarget: 'var'` (https://webpack.js.org/configuration/output/#expose-a-variable) and remove the `jsonpFunction`

Comment (AviVahl) [reply packages/scripts/src/create-webpack-configs.ts:280]: will try

Comment (RomanYarik) [reply packages/scripts/src/create-webpack-configs.ts]: @AviVahl seems like removing the libraryTarget whatsoever fixed the issue, and now tests pass

**Comment by AviVahl** [comment]
> no need for single run anymore? how do you make the webpackDevMiddeware to "not watch" for current run?
was just a hack for the days it crashed on mac when watching.

**Comment by AviVahl** [comment]
Thanks for figuring it out. I basically tried everything other than just removing it. XD

**Comment by AviVahl** [comment]
Now, while this passed tests, I'm not really sure it's still compatible with `webpack@4`. I'll need to test this further and make fixes to get it *really* working with both versions

File: packages/scripts/src/create-webpack-configs.ts [hunk 4]
```diff
@@ -291,14 +291,19 @@ export function createWebpackConfigForExternalFeature({
         },
         stats: 'errors-warnings',
     };
+    if (semverLessThan(webpack.version, '5.0.0')) {
```
Comment (AviVahl) [inline packages/scripts/src/create-webpack-configs.ts:294]: this was just added in the last rebase.
will make it compatible with both 4.0.0 and 5.0.0.
